internal/graph/check.go

```diff
@@ -372,6 +372,12 @@ func (c *LocalChecker) checkDirect(parentctx context.Context, req *ResolveCheckR
 		storeID := req.GetStoreID()
 		tk := req.GetTupleKey()
 
+		var allowedTypesForUser []*openfgapb.RelationReference
+		if typesys.GetSchemaVersion() == typesystem.SchemaVersion1_1 {
+			// allowedTypesForUser could be "user" or "user:*" or "group#member"
+			allowedTypesForUser, _ = typesys.GetDirectlyRelatedUserTypes(tuple.GetType(tk.GetObject()), tk.Relation)
```

Comment:
Please create a closure around these variables since we use them in multiple places in both `fn1` and `fn2`.

```
objectType := tuple.GetType(tk.GetObject())
relation := tk.GetRelation()

fn1 := func(ctx conetxt.Context) ... {
    ..
    typesystem.DirectRelationReference(objectType, relation),
    ...
}

fn2 := func(ctx context.Context) ... {
    // use 'objectType' and 'relation'
}
```